pkg/resources.go

```diff
@@ -77,21 +78,26 @@ func (r Resource) GetPropertiesMap() map[string]string {
 func (r Resource) GeneratePropertesMapFromProperties() map[string]string {
 	var propertiesMap = make(map[string]string)
 	var prop string
-	if r.Item.GetKind() == "Secret" {
-		val, _ := base64.Decode(r.Item.Object["data"].(map[string]interface{})[r.Hierarchy.Key].(string))
-		prop = string(val)
-	} else {
-		prop = r.Item.Object["data"].(map[string]interface{})[r.Hierarchy.Key].(string)
-	}
-	for _, keyValue := range strings.Split(prop, "\n") {
-		propKeyValue := strings.Split(keyValue, "=")
-		if len(propKeyValue) == 2 {
-			propKey := propKeyValue[0]
-			propValue := propKeyValue[1]
-			propertiesMap[propKey] = propValue
+	value := r.Item.Object["data"].(map[string]interface{})[r.Hierarchy.Key]
+	if value != nil {
+		if r.Item.GetKind() == "Secret" {
+			val, _ := base64.Decode(value.(string))
+			prop = string(val)
+		} else {
+			prop = value.(string)
+		}
+		for _, keyValue := range strings.Split(prop, "\n") {
+			propKeyValue := strings.Split(keyValue, "=")
+			if len(propKeyValue) == 2 {
+				propKey := propKeyValue[0]
+				propValue := propKeyValue[1]
+				propertiesMap[propKey] = propValue
+			}
 		}
+		return propertiesMap
 	}
-	return propertiesMap
+	logger.Errorf("can not find the given key in the configMap %v", r.Hierarchy.Key)
+	return nil
```

Comment:
Can do:

```go


rather exit early, then put the main flow in a if statement
Also don't log an error - either return it as an error - If it is an acceptable condition, then log in debug
```go 
if value == nil {
  	logger.Errorf("can not find the given key in the configMap %v", r.Hierarchy.Key)
	return nil
}
```